src/vs/editor/common/controller/cursorTypeOperations.ts

```diff
@@ -498,31 +498,30 @@ export class TypeOperations {
 		});
 	}
 
-	private static _autoClosingPairIsSymmetric(autoClosingPair: StandardAutoClosingPairConditional): boolean {
-		const { open, close } = autoClosingPair;
-		return (open.indexOf(close) >= 0 || close.indexOf(open) >= 0);
-	}
+	private static _isBeforeClosingBrace(config: CursorConfiguration, lineAfter: string) {
+		// If the start of lineAfter can be interpretted as both a starting or ending brace, default to returning false
+		const nextChar = lineAfter.charAt(0);
 
-	private static _isBeforeClosingBrace(config: CursorConfiguration, autoClosingPair: StandardAutoClosingPairConditional, characterAfter: string) {
-		const otherAutoClosingPairs = config.autoClosingPairsClose2.get(characterAfter);
-		if (!otherAutoClosingPairs) {
-			return false;
+		// check that lineAfter does NOT immediately start with a starting brace
+		const potentialStartingBraces = config.autoClosingPairs.autoClosingPairsCloseByStart.get(nextChar) || [];
+		for (const startingBrace of potentialStartingBraces) {
+			if (lineAfter.startsWith(startingBrace.open)) {
+				return false;
+			}
 		}
 
-		const thisBraceIsSymmetric = TypeOperations._autoClosingPairIsSymmetric(autoClosingPair);
-		for (const otherAutoClosingPair of otherAutoClosingPairs) {
-			const otherBraceIsSymmetric = TypeOperations._autoClosingPairIsSymmetric(otherAutoClosingPair);
-			if (!thisBraceIsSymmetric && otherBraceIsSymmetric) {
-				continue;
+		// check that lineAfter DOES immediately starts with a closing brace
+		const potentialClosingBraces = config.autoClosingPairs.autoClosingPairsCloseByStart.get(nextChar) || [];
```

Comment:
Lines 507-510 almost duplicate these lines.  One could move to a help function and pass in the potential array and start character to check for.

Reply:
True, though a helper function would also need a bool flag for checking open/close.

I just checked and it turns out javascript does actually have built in `Array.prototype.some`, which is all it's really doing anyway.

How would you feel about something like this?

```
private static _isBeforeClosingBrace(config: CursorConfiguration, lineAfter: string) {
    // If the start of lineAfter can be interpretted as both a starting or ending brace, default to returning false
    const nextChar = lineAfter.charAt(0);
    const potentialStartingBraces = config.autoClosingPairs.autoClosingPairsOpenByStart.get(nextChar) || [];
    const potentialClosingBraces = config.autoClosingPairs.autoClosingPairsCloseByStart.get(nextChar) || [];

    const isBeforeStartingBrace = potentialStartingBraces.some(x => lineAfter.startsWith(x.open));
    const isBeforeClosingBrace = potentialClosingBraces.some(x => lineAfter.startsWith(x.close));

    return !isBeforeStartingBrace && isBeforeClosingBrace;
}
```

Reviewing that made me realise L507 should technically be autoClosingPairs**Open**ByStart

Reply:
This looks good to me.

Reply:
Thanks for the review :)